main/src/main/scala/sbt/Defaults.scala

```diff
@@ -383,6 +383,7 @@ object Defaults extends BuildCommon {
       },
       serverHandlers :== Nil,
       windowsServerSecurityLevel := Win32SecurityLevel.OWNER_DACL, // allows any owner logon session to access the server
+      serverUseJna := !BootServerSocket.supportsJNI || SysProp.serverUseJna,
```

Comment:
Given the maintenance burden of JNI, I think we should use JNA by default and battle test it.
JNI requires building C code on all platforms we plan to support, whereas JNA binding is available for dozens of OS + arch - https://github.com/java-native-access/jna/tree/master/lib/native.

Reply:
I disagree. Jna is more complicated internally and we can’t do anything about it when it breaks like it did here with the new apple silicon. We are only using JNI on the same platforms where we have to use JNI for the thin client anyway. It isn’t clear to me that JNA will ever work with the thin client so we already have a permanent maintenance burden on using the JNI with ipcsocket. By making it the default on the server, we also get more battle testing so if there is any discrepancy between the jni and jna implementation that we can fix it.

Reply:
That all being said, is the suggestion that we only default to jni on arm64 macs?

Reply:
One last clarification, the reason I wanted to make jni default where supported is that I don’t think #6062 would have happened if jni had been the default since the apple silicon can supposedly run intel binaries. I suppose this is unusual because apple rarely changes platforms but it seems possible that microsoft could do something similar with a new windows release. At the end of the day, I guess it isn’t a big deal if we have to release a new sbt to support this kind of os upgrade but it would be nice if we didn’t have to.

Reply:
And just to be clear, regardless of how we set the default on supported jni platforms, we will always fallback to jna on unsupported platforms which is why it’s great that we have both jni and jna implementations.

Reply:
> That all being said, is the suggestion that we only default to jni on arm64 macs?

Yes.

My thinking here is that I'm bullish about thin client becoming the _only_ client eventually. In that future, we need to have the socket working in order to use sbt in Solaris etc, and those users can use pure-Java implementation of the thin client, which already exists thanks to you implementing it in JVM.

Reply:
I don't see how these goals are in conflict but I also don't want to continue this debate so I just went ahead and made the suggested change.